### ISSUE TITLE
generalized resource class

### ISSUE THREAD

**Issue by 7yl4r** [7yl4r/the-oregon-trajectory#51]
before adding more resources, a resource class ought to be use to reduce duplication

```coffeescript
class resource
    constructor: ()->
        @chance = 0.5  # chance of use per tick
        @expense = 1   # amount subtracted per use
        @amount   = 10 # how much you currently have

    tick: ()->
        if Math.random() < @chance
            @amount -= @expense
```

**Comment by BrianErikson** [comment]
I'm not sure I understand the chance variable. I mean, it makes sense with radiation and other random things, but since a 'tick' is basically just time progressing, does it make sense to have a "chance" to expend fuel? I'm curious :)

**Comment by 7yl4r** [comment]
I guess I'm thinking that it's a little fun to have some uncertainty in there so that sometimes you get lucky. It seems like it's not very noticeable as implemented though, with as many times as we're `random`ing it's virtually the same as `* @chance`

**Comment by BrianErikson** [comment]
Yeah I agree with that. Maybe a tick every 250ms or something? Four a second? Or we could even go more old-school with it and do two a second

**Comment by 7yl4r** [comment]
before/during changing that (and balancing the other constants) might be a good time to address #53 too.
